addons/pos_restaurant/static/src/overrides/components/navbar/navbar.js

```diff
@@ -12,6 +12,9 @@ patch(Navbar, {
 patch(Navbar.prototype, {
     async onClickBackButton() {
         if (this.pos.mainScreen.component && this.pos.config.module_pos_restaurant) {
+            if (!this.pos.selectedTable) {
+                this.pos.add_new_order();
+            }
```

Comment:
I think this isn't the right place to fix the issue. It should be in the `unsetTable`.
You might need to set `isBooked` to true for free orders.

https://github.com/odoo/odoo/blob/4ac250c1544f16930a5b108e836f4588f28f9309/addons/pos_restaurant/static/src/overrides/models/pos_store.js#L301